src/xSTIR/pSTIR/STIR.py.in

```diff
@@ -722,9 +722,18 @@ class AcquisitionData(DataContainer):
         - number of tangential positions.
         '''
         assert self.handle is not None
-        array = numpy.ndarray(self.dimensions(), dtype = numpy.float32)
-        try_calling(pystir.cSTIR_getAcquisitionData\
-            (self.handle, array.ctypes.data))
+        # If using file storage scheme, create numpy array and copy data
+        if self.get_storage_scheme() == "file":
+            array = numpy.ndarray(self.dimensions(), dtype = numpy.float32)
+            try_calling(pystir.cSTIR_getAcquisitionData\
+                (self.handle, array.ctypes.data))
+        # If using memory, use a buffer and have a shared object
+        else:
+            buffer = ?
+            try_calling(pystir.cSTIR_AcquisitionDataInMemory_get_buffer\
+                (self.handle, buffer))
+            array = numpy.frombuffer(buffer, dtype=numpy.float32)
```

Comment:
I do not see how this can possibly work. However you define `buffer`, `pystir.cSTIR_AcquisitionDataInMemory_get_buffer` can only give you an address in memory, but `numpy.frombuffer` needs to know the size of memory represented by the buffer to be able to handle that memory as numpy array data.

Reply:
I haven't done it here, but you can give the number of elements with `count`:
```
numpy.frombuffer(buffer, dtype=float, count=-1, offset=0)
```
And obviously we know how many elements our sinogram contains.

Reply:
The problem is that the first argument to `frombuffer` should be of type `buffer_like` (see [here](https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html)). I have no idea how to do this, but the address to the first float in C++ somehow needs to be converted to this `buffer_like` object in Python. Searching online, it looks like it's possible with SWIG, but I'm not sure how to do it in SIRF's setup.